internal/testplanet/satellite.go

```diff
@@ -234,10 +391,87 @@ func (planet *Planet) newSatellites(count int) ([]*SatelliteSystem, error) {
 		planet.databases = append(planet.databases, db)
 		planet.databases = append(planet.databases, pointerDB)
 
-		log.Debug("id=" + peer.ID().String() + " addr=" + peer.Addr())
+		api, err := planet.newAPI(i, identity, db, pointerDB, config, versionInfo)
+		if err != nil {
+			return xs, err
+		}
+		log.Debug("id=" + peer.ID().String() + " addr=" + api.Addr())
 
-		system := SatelliteSystem{Peer: *peer}
-		xs = append(xs, &system)
+		system := createNewSystem(log, peer, api)
+		xs = append(xs, system)
 	}
 	return xs, nil
 }
+
+// createNewSystem makes a new Satellite System and exposes the same interface from
+// before we split out the API. In the short term this will help keep all the tests passing
+// without much modification needed. However long term, we probably want to rework this
+// so it represents how the satellite will run when it is made up of many prrocesses.
+func createNewSystem(log *zap.Logger, peer *satellite.Peer, api *satellite.API) *SatelliteSystem {
```

Comment:
could we specify that this relates to the satellite? Like `newSatelliteSystem`?

Reply:
oh, well since it's in the `satellite.go` file it should be okay. Feel free to ignore